### ISSUE TITLE
Roland MT32 suport for Amiberry

### ISSUE THREAD

**Issue by ransom1122** [BlitterStudio/amiberry#420]
Hi Midwan.

In WinUAE if you go to IO ports and select MT-32 Synth Emulator as MIDI out. You can I play amiga games with Roland sound which sound amazing.

Is there any possible way this is or can be incorporated with Amiberry?

ransom

**Comment by ransom1122** [comment]
It can be installed on pie, just need the ports activation In Amiberry if possible.

https://retropie.org.uk/forum/topic/12549/tutorial-installing-munt-mt-32-emulation-on-rpi-3

**Comment by 5schatten** [comment]
IMHO you should have a look at [FluidSynth](https://github.com/FluidSynth/fluidsynth) or [Munt MT-32](https://github.com/munt/munt/blob/master/DOSBox-mt32-patch/dosbox-SVN-r4025-mt32-patch.diff)

Maybe the way Retroarch implemented an MIDI driver could give a hint. https://github.com/libretro/RetroArch/commit/c1662ded34792b3e8746292ad573640b2191c8e9

Both works fine for me with DOSBox & Retroarch. The MT-32 needs some blobs while FluidSynth depends on soundfonts. Either way the sound is superior to non MIDI music ;-)

**Comment by midwan** [comment]
We could look into this in the future, but I think we need to give higher priority to some of the other items in the list.

**Comment by 2play** [comment]
I d like to see mt32 support on amiberry. 
it would be nice to have in GUI the option to set mt32 as its done in scummvm.
select mix, soundfont, and mt32 emulation

**Comment by joolswills** [comment]
@2play That's a joke right? For what purpose? :-)

**Comment by joolswills** [comment]
I didn't read the whole thread sorry. Have now but I'm confused. Amiga games didn't generally use Midi. @ransom1122 which games?

I understand soundcard support like toccata etc.

**Comment by 2play** [comment]
@joolswills  here are some that have mt32. as @ransom1122 said this is not just supported on amiberry yet.
but if same setting/handling as on scummvm enabled then I belive these mt32 can be used.
I havent used MUNT so i dont know if that workaround might work

**Comment by 2play** [comment]
@joolswills i hope by now that im not making jokes of that kind. I know @midwan in a way(not in peson )and have great respect for him and his work.
good day to all

**Comment by midwan** [comment]
This is now implemented in the `feature/mt32` branch, in case anyone wants to test it. I did a quick test with Castle of Dr. Brain (MT32 version) and it seems to work as expected.

It requires the MT32 ROMs, placed inside the `kickstarts/mt32-roms/` directory - that is hardcoded for now.
The branch can only be built with CMake for now, the Makefile will be updated later.

If they are not found, the dropdown will notify you accordingly...

![image](https://github.com/BlitterStudio/amiberry/assets/3877365/94cc5b75-0b24-4314-b848-d6551c30d8ea)
